### PR TITLE
Changed 5 events into transpilers

### PR DESCRIPTION
**Pull request by iopietro**
+ Added ElevatorTeleportEventArgs::AuxiliaryPowerCost and made obsolete ElevatorTeleportEventArgs::APCost.
+ Added InteractingDoorEventArgs::AuxiliaryPowerCost.
+ Added InteractingTeslaEventArgs::AuxiliaryPowerCost.
+ Added StartingSpeakerEventArgs::AuxiliaryPowerCost and made obsolete StartingSpeakerEventArgs::APDrain.

### CODE REVIEW

**Comment by iopietro** [comment]
If you don't like `InteractingDoorEventArgs::AuxiliaryPowerCost` because it's being used by players as well, we could change that into a new event which will inherit from `InteractingDoorEventArgs`.

Also, I don't know why `ElevatorTeleportEventArgs` had been approved, since it doesn't respect our events naming convention (`-ing` for pre-events and `-ed` for post-events).

**Comment by iRebbok** [comment]
> If you don't like `InteractingDoorEventArgs::AuxiliaryPowerCost` because it's being used by players as well, we could change that into a new event which will inherit from `InteractingDoorEventArgs`.

Yes, please. It'll seem complicated with additional unused variables.

> Also, I don't know why `ElevatorTeleportEventArgs` had been approved since it doesn't respect our events naming convention (`-ing` for pre-events and `-ed` for post-events), it should have been named something like `ElevatorTeleportingEventArgs`.

I agree with such a change, it makes sense.

**Comment by iopietro** [comment]
Done.

File: Exiled.Events/EventArgs/ElevatorTeleportEventArgs.cs [hunk 1]
```diff
@@ -42,9 +42,19 @@ public ElevatorTeleportEventArgs(Player player, Camera079 camera, float apCost,
         public Camera079 Camera { get; set; }
 
         /// <summary>
-        /// Gets or sets the amount of AP will be consumed during the level change.
+        /// Gets or sets the amount of auxiliary power required to interact with a tesla gate.
```
Comment (iRebbok) [inline Exiled.Events/EventArgs/ElevatorTeleportEventArgs.cs:45]: It's not about interacting with a test gate.

File: Exiled.Events/EventArgs/ElevatorTeleportEventArgs.cs [hunk 2]
```diff
@@ -42,9 +42,19 @@ public ElevatorTeleportEventArgs(Player player, Camera079 camera, float apCost,
...
         /// </summary>
-        public float APCost { get; set; }
+        public float AuxiliaryPowerCost { get; set; }
+
+        /// <summary>
+        /// Gets or sets the amount of AP that will be removed for the first time when using speakers through SCP-079.
```
Comment (iRebbok) [inline Exiled.Events/EventArgs/ElevatorTeleportEventArgs.cs:50]: Ugh...

File: Exiled.Events/EventArgs/ElevatorTeleportingEventArgs.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        /// </summary>
+        public Player Player { get; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="Camera079"/> that SCP-079 will be moved to.
+        /// </summary>
+        public Camera079 Camera { get; set; }
+
+        /// <summary>
+        /// Gets or sets the amount of auxiliary power required to interact with a tesla gate.
```
Comment (iRebbok) [inline Exiled.Events/EventArgs/ElevatorTeleportingEventArgs.cs:45]: oof

File: Exiled.Events/EventArgs/ElevatorTeleportingEventArgs.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        /// </summary>
+        public float AuxiliaryPowerCost { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not SCP-079 can teleport.
+        /// Defaults to a <see cref="bool"/> describing whether or not SCP-079 has enough AP to teleport.
```
Comment (iRebbok) [inline Exiled.Events/EventArgs/ElevatorTeleportingEventArgs.cs:51]: Wouldn't it be better to replace AP with auxiliary power?

File: Exiled.Events/EventArgs/TriggeringDoorEventArgs.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        /// <param name="auxiliaryPowerCost"><inheritdoc cref="AuxiliaryPowerCost"/></param>
+        /// <param name="isAllowed"><inheritdoc cref="InteractingDoorEventArgs.IsAllowed"/></param>
+        public TriggeringDoorEventArgs(Player player, DoorVariant door, float auxiliaryPowerCost, bool isAllowed = true)
+            : base(player, door, isAllowed)
+        {
+            AuxiliaryPowerCost = auxiliaryPowerCost;
+        }
+
+        /// <summary>
+        /// Gets or sets the amount of auxiliary power required to interact with a tesla gate.
```
Comment (iRebbok) [inline Exiled.Events/EventArgs/TriggeringDoorEventArgs.cs:33]: 👀

**Review by iRebbok — CHANGES_REQUESTED** [review]

File: Exiled.Events/EventArgs/StartingSpeakerEventArgs.cs [hunk 6]
```diff
@@ -41,10 +41,20 @@ public StartingSpeakerEventArgs(Player player, Room room, float apDrain, bool is
         /// </summary>
         public Room Room { get; }
 
+        /// <summary>
+        /// Gets or sets the amount of auxiliary power required to interact with a tesla gate.
```
[the remark this answers is not in the document]
Comment (iopietro) [reply Exiled.Events/EventArgs/StartingSpeakerEventArgs.cs:45]: Ah yeah, I copy pasted that property for all events and I've forgot to change the description.

File: Exiled.Events/EventArgs/ElevatorTeleportingEventArgs.cs (hunk 4 above)
Comment (iopietro) [reply Exiled.Events/EventArgs/ElevatorTeleportingEventArgs.cs:51]: Sure, I'll change that comment (I copy pasted it from the old event).

File: Exiled.Events/EventArgs/ElevatorTeleportingEventArgs.cs (hunk 3 above)
Comment (iopietro) [reply Exiled.Events/EventArgs/ElevatorTeleportingEventArgs.cs:45]: Apparently, everything is a tesla gate, lol.

File: Exiled.Events/EventArgs/ElevatorTeleportEventArgs.cs [hunk 7]
```diff
@@ -21,13 +21,13 @@ public class ElevatorTeleportEventArgs : EventArgs
         /// </summary>
         /// <param name="player"><inheritdoc cref="Player"/></param>
         /// <param name="camera"><inheritdoc cref="Camera"/></param>
-        /// <param name="apCost"><inheritdoc cref="APCost"/></param>
+        /// <param name="auxiliaryPowerCost"><inheritdoc cref="AuxiliaryPowerCost"/></param>
         /// <param name="isAllowed"><inheritdoc cref="IsAllowed"/></param>
-        public ElevatorTeleportEventArgs(Player player, Camera079 camera, float apCost, bool isAllowed = true)
+        public ElevatorTeleportEventArgs(Player player, Camera079 camera, float auxiliaryPowerCost, bool isAllowed = true)
```
[the remark this answers is not in the document]
Comment (iopietro) [reply Exiled.Events/EventArgs/ElevatorTeleportEventArgs.cs:26]: I could do it.

File: Exiled.Events/EventArgs/ElevatorTeleportEventArgs.cs (hunk 2 above)
Comment (iopietro) [reply Exiled.Events/EventArgs/ElevatorTeleportEventArgs.cs:50]: Do I have to change that description? It's an obsolete property after all.

Comment (iRebbok) [reply Exiled.Events/EventArgs/ElevatorTeleportEventArgs.cs]: I can say yeah, even if a thing is deprecated, we should maintain full documentation for it.

**Review by iRebbok — APPROVED** [review]

**Review by galaxy119 — APPROVED** [review]

**Pull request merged**